src/c++/vdo/base/dm-vdo-target.c

```diff
@@ -1068,10 +1068,30 @@ static void vdo_io_hints(struct dm_target *ti, struct queue_limits *limits)
 	limits->logical_block_size = vdo->device_config->logical_block_size;
 	limits->physical_block_size = VDO_BLOCK_SIZE;
 
+#ifndef VDO_UPSTREAM
+#undef VDO_USE_ALTERNATE
+#if defined(RHEL_RELEASE_CODE) && defined(RHEL_MINOR) && (RHEL_MINOR < 50)
+#if (RHEL_RELEASE_CODE < RHEL_RELEASE_VERSION(10, 0))
+#define VDO_USE_ALTERNATE
+#endif
+#else /* !RHEL_RELEASE_CODE */
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 12, 0))
```

Comment:
This code actually got pulled into 6.11 as far as I can tell. So this boundary is not quite right.

But Susan made a good point that the limits structure has been there for a while, and we can probably set this unilaterally, without using VDO_USE_ALTERNATE.